src/main.js

```diff
@@ -22,11 +22,23 @@ const checkValidationOfValues = envSpecString => {
   //element[0] is the variable e.g. ADMIN_EMAIL
   //element[1] is the type e.g. test
   envValArray = envValArray.filter(element => {
-    //check for valid variables AND types
+    //check for valid variables AND types or restricted choices
     if (
       element[0].match(alphanumericThatDoesNotStartWithDigit) &&
       validTypes.includes(element[1])
     ) {
+      //in case value has valid type
+      element[2] = 0; //0 constant that declares what should be printed for an element with type
+      return element;
+    }
+    // in case value has restricted choices (indicated by "[]" ,we should split them
+    else if (
+      element[0].match(alphanumericThatDoesNotStartWithDigit) &&
+      element[1][0] === "[" &&
```

Comment:
I see what you are doing here, but it's quite crude and involves multiple checks. We should prefer using a regular expression and extract the options inside:

- About Regular Expressions: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions
- Extracting information with parentheses: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#Using_parenthesized_substring_matches